torch/distributed/distributed_c10d.py

```diff
@@ -572,6 +573,21 @@ class GroupMember(metaclass=_WorldMeta):
 
 STORE_BASED_BARRIER_PREFIX = "store_based_barrier_key"
 
+# DO NOT USE this OBJECT DIRECTLY.
+# Use the APIs below instead.
+_default_backend_for_device: Dict[str, Backend] = {
```

Comment:
@H-Huang what's the canonical way to register a backend for a device type in c10d?

Reply:
@wanchaol I've removed this change. but i think we need to sync on how `DeviceMesh` will create process groups for a non-nccl and non-gloo backend.